include/TaggedUnion.h

```diff
@@ -117,3 +117,9 @@ struct TaggedUnion {
     }\
   };
 
+
+#define _tagged_union_join2(a, b) a ## b
+#define _tagged_union_join(a, b) _tagged_union_join2(a, b)
```

Comment:
Out of curiosity, why is this pattern so common?  What case isn't handled by just a single `#define _tagged_union_join(a, b) a ## b` macro?

Reply:
it forces the arguments to be evaluated.  If you only had 1 macro, then it wouldn't resolve `__LINE__` to the actual line number.